### PR TITLE
add tests to ensure functions preserve comments

### PR DESCRIPTION
**Pull request by mengqiy**
closes: https://github.com/GoogleContainerTools/kpt/issues/1815

### CODE REVIEW

File: examples/format/imperative/.expected/diff.patch [hunk 1]
```diff
@@ -1,37 +1,40 @@
...
  kind: Deployment
  metadata:
 -  labels:
 -    foo: bar
--    env: dev
+-    env: dev # must be one of prod, staging and dev
    name: my-nginx
+   # namespace for team foo
    namespace: my-space
 +  labels:
```
Comment (frankfarzan) [inline examples/format/imperative/.expected/diff.patch:15]: can you explain the formatting changes in the PR description.

Comment (mengqiy) [reply examples/format/imperative/.expected/diff.patch:15]: This diff is from the `format` function. The formatting change is desired.
And also this line is not changed in this PR.

**Comment by phanimarupaka** [comment]
@mengqiy what was the root cause and how is this fixed in code ?

**Comment by mengqiy** [comment]
Closing in favor of https://github.com/GoogleContainerTools/kpt/pull/2169, we are preserving the comments in the orchestrator layer

**Pull request closed without merging**